### ISSUE TITLE
Ghostrunner 2

### ISSUE THREAD

**Issue by ImASadBoy** [Nukem9/dlssg-to-fsr3#106]
Hello! Ghostrunner 2 crashes on activation of FG with following error:

https://imgur.com/a/vtNRcKO

**Comment by Gaffer13** [comment]
And DLSSTweaks version doesn't do anything. You get the dialog at launch, can toggle frame generations menu slider when you disable, then enable DLSS - but it doesn't generate frames.

**Comment by ImASadBoy** [comment]
@Gaffer13 DLSS Tweaks doesn't give the pop-up to me

**Comment by Gaffer13** [comment]
Have you edited the path to DLSSG file in 

`Z:\Ghostrunner 2\Ghostrunner2\Binaries\Win64`

Have you put the files there in the first place?

**Comment by ImASadBoy** [comment]
Yes, I have done both

**Comment by doktorsleepelss** [comment]
I get the pop up and it doesn't crash for me.  But the frame gen toggle is greyed out. You can ungrey it by turning off dlss and turning it back on, but turning it still doesn't do anything. And if you go back to the settings, it'll be greyed out again.

**Comment by Gaffer13** [comment]
Same for me.

**Comment by ImASadBoy** [comment]
Problem still persists with v0.60

**Comment by Lennn888** [comment]
Works here. If frame gen is greyed out, make sure you launched game with -dx12 launch argument as it is a requirement.
If it crashes when enabled, open _GameUserSettings.ini_ in %localappdata%\Ghostrunner2\Saved\Config\WindowsNoEditor change "DLSSG=0" to "DLSSG=1", save the file and run the game.

**Comment by ImASadBoy** [comment]
> Works here. If frame gen is greyed out, make sure you launched game with -dx12 launch argument as it is a requirement. If it crashes when enabled, open _GameUserSettings.ini_ in %localappdata%\Ghostrunner2\Saved\Config\WindowsNoEditor change "DLSSG=0" to "DLSSG=1", save the file and run the game.

Yes I already tried to force it via the .ini amd set the ini read-only. Problem is: the FG is on and looks on but is not generating any frame, try toggoling it off, you'll see no frametate difference.

**Comment by Gaffer13** [comment]
I forced DX12, but now I'm getting "DXGI ERROR INVALID CALL" when I try to enable frame generation.

**Comment by Gaffer13** [comment]
Still crashes with 0.70

**Comment by ImASadBoy** [comment]
ACtually now you can turn on FG without the game crashing, but the screen shows a static black image (sounds still work tho)

**Comment by xCaptainTroy** [comment]
now with the latest version the FG works perfect

**Comment by Gaffer13** [comment]
Works for me too - but performance in DX12 is often much worse than in DX11, so there's no point in switching to DX12 only to use frame generation. More importantly, I'm getting serious performance degradation after a while - as if the game is running out of VRAM. 

RTX2060 6GB, "Setting the stage" - even going back to a mid-level checkpoint after quitting the game, performance slows to a crawl in a couple of minutes. Doesn't happen without the mod, so might be a memory leak, not just low VRAM.

**Comment by xCaptainTroy** [comment]
> Works for me too - but performance in DX12 is often much worse than in DX11, so there's no point in switching to DX12 only to use frame generation. More importantly, I'm getting serious performance degradation after a while - as if the game is running out of VRAM.
> 
> RTX2060 6GB, "Setting the stage" - even going back to a mid-level checkpoint after quitting the game, performance slows to a crawl in a couple of minutes. Doesn't happen without the mod, so might be a memory leak, not just low VRAM.

I confirm that isssue but only happened to me at 4k full maxed out, but at 1080 full maxed out + ray tracing DLSS Q + FG I got better performance than dx 11 280~310 FPS

**Comment by Gaffer13** [comment]
My point was, _baseline_ performance is worse in DX12. It's still as good as, or better _after_ FG - but in effect you're trading some real frames for generated ones. 

Of course, if you want raytracing, you have to use DX12.

**Comment by ImASadBoy** [comment]
Confirming it works! I updated the wiki page for the game, the VRAM crash thing only happens with Raytracing ON since Raytracing is currently not officially supported by the developers , looks like it doesn't flush the memory so the longer you play the heavier it will be on VRAM.

**Comment by Gaffer13** [comment]
> Confirming it works! I updated the wiki page for the game, the VRAM crash thing only happens with Raytracing ON since Raytracing is currently not officially supported by the developers , looks like it doesn't flush the memory so the longer you play the heavier it will be on VRAM.

I've had issues (not a crash) even with raytracing off. The "base" seems fine, but when I get into the (rather large) "Setting the stage" level, I get the slowdown. How much VRAM do you have? Are you using the DLSSTweaks version?

**Comment by ImASadBoy** [comment]
> > Confirming it works! I updated the wiki page for the game, the VRAM crash thing only happens with Raytracing ON since Raytracing is currently not officially supported by the developers , looks like it doesn't flush the memory so the longer you play the heavier it will be on VRAM.
> 
> I've had issues (not a crash) even with raytracing off. The "base" seems fine, but when I get into the (rather large) "Setting the stage" level, I get the slowdown. How much VRAM do you have? Are you using the DLSSTweaks version?

The issue is not the mod is the game in DX12, that's why developers disabled it. I'd say wait for patches to officially support it.

**Comment by Gaffer13** [comment]
Like I said, I tried removing the mod and loading the same checkpoint - even if the issue is with the game, the mod might be making it worse. I'll try other games, of course.

**Comment by Gaffer13** [comment]
Looks like the game has performance degradation even in DX11 and even without the mod.
